From ebc5649d4e5320233744b35ecee2156a34ac6681 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Thu, 20 Apr 2023 13:35:35 +0300
Subject: [PATCH] lx2162som: update clearfog configurations for latest MC
 firmware

Add compatible properties to all dpl nodes.
Also drop some unused serdes protocol combinations.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 .../LX2162-USOM/clearfog-s1_0-s2_11-dpc.dts   | 102 -------------
 .../LX2162-USOM/clearfog-s1_0-s2_11-dpl.dts   | 142 ------------------
 .../LX2162-USOM/clearfog-s1_0-s2_7-dpc.dts    | 102 -------------
 .../LX2162-USOM/clearfog-s1_0-s2_7-dpl.dts    | 142 ------------------
 .../LX2162-USOM/clearfog-s1_3-s2_0-dpc.dts    |  94 ------------
 .../LX2162-USOM/clearfog-s1_3-s2_0-dpl.dts    | 138 -----------------
 ...2_9-dpc.dts => clearfog-s1_3-s2_7-dpc.dts} |  21 ++-
 ...2_9-dpl.dts => clearfog-s1_3-s2_7-dpl.dts} | 108 ++++++++++++-
 .../LX2162-USOM/clearfog-s1_3-s2_9-dpl.dts    |  98 +++++++++++-
 9 files changed, 212 insertions(+), 735 deletions(-)
 delete mode 100644 config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_11-dpc.dts
 delete mode 100644 config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_11-dpl.dts
 delete mode 100644 config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_7-dpc.dts
 delete mode 100644 config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_7-dpl.dts
 delete mode 100644 config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_0-dpc.dts
 delete mode 100644 config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_0-dpl.dts
 rename config/lx2160a/LX2162-USOM/{clearfog-s1_0-s2_9-dpc.dts => clearfog-s1_3-s2_7-dpc.dts} (94%)
 rename config/lx2160a/LX2162-USOM/{clearfog-s1_0-s2_9-dpl.dts => clearfog-s1_3-s2_7-dpl.dts} (64%)

diff --git a/config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_11-dpc.dts b/config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_11-dpc.dts
deleted file mode 100644
index b199f64..0000000
--- a/config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_11-dpc.dts
+++ /dev/null
@@ -1,102 +0,0 @@
-/*
- * Copyright 2020 NXP
- * Copyright 2022 Josua Mayer <josua@solid-run.com>
- *
- * Redistribution and use in source and binary forms, with or without
- * modification, are permitted provided that the following conditions are met:
- *   * Redistributions of source code must retain the above copyright
- *     notice, this list of conditions and the following disclaimer.
- *   * Redistributions in binary form must reproduce the above copyright
- *     notice, this list of conditions and the following disclaimer in the
- *     documentation and/or other materials provided with the distribution.
- *   * Neither the name of the above-listed copyright holders nor the
- *     names of any contributors may be used to endorse or promote products
- *     derived from this software without specific prior written permission.
- *
- * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
- * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
- * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
- * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
- * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
- * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
- * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
- * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
- * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
- * POSSIBILITY OF SUCH DAMAGE.
- */
-
-/dts-v1/;
-
-/ {
-	resources {
-		icid_pools {
-			icid_pool@1 {
-				num = <0x64>;
-				base_icid = <0x0>;
-			};
-		};
-	};
-
-	mc_general {
-		log {
-			mode = "LOG_MODE_ON";
-			level = "LOG_LEVEL_WARNING";
-		};
-
-		console {
-			mode = "CONSOLE_MODE_OFF";
-			uart_id = <0x4>;
-			level = "LOG_LEVEL_WARNING";
-		};
-	};
-
-	controllers {
-		qbman {
-			/* Transform this number of 8-WQ channels into four times
-			 * as many 2-WQ channels. This allows the creation of a
-			 * larger number of DPCONs.
-			 */
-			wq_ch_conversion = <64>;
-		};
-	};
-
-	board_info {
-		recycle_ports {
-			recycle@1 {
-				max_rate = "1G";
-			};
-
-			recycle@2 {
-				max_rate = "1G";
-			};
-		};
-
-		ports {
-			/* Serdes 2 */
-			mac@12 {
-				link_type = "MAC_LINK_TYPE_PHY";
-			};
-
-			mac@13 {
-				link_type = "MAC_LINK_TYPE_PHY";
-			};
-
-			mac@14 {
-				link_type = "MAC_LINK_TYPE_PHY";
-			};
-
-			mac@16 {
-				link_type = "MAC_LINK_TYPE_PHY";
-			};
-
-			mac@17 {
-				link_type = "MAC_LINK_TYPE_PHY";
-			};
-
-			mac@18 {
-				link_type = "MAC_LINK_TYPE_PHY";
-			};
-		};
-	};
-};
diff --git a/config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_11-dpl.dts b/config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_11-dpl.dts
deleted file mode 100644
index 766b04d..0000000
--- a/config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_11-dpl.dts
+++ /dev/null
@@ -1,142 +0,0 @@
-/*
- * Copyright 2020 NXP
- * Copyright 2022 Josua Mayer <josua@solid-run.com>
- *
- * Redistribution and use in source and binary forms, with or without
- * modification, are permitted provided that the following conditions are met:
- *   * Redistributions of source code must retain the above copyright
- *     notice, this list of conditions and the following disclaimer.
- *   * Redistributions in binary form must reproduce the above copyright
- *     notice, this list of conditions and the following disclaimer in the
- *     documentation and/or other materials provided with the distribution.
- *   * Neither the name of the above-listed copyright holders nor the
- *     names of any contributors may be used to endorse or promote products
- *     derived from this software without specific prior written permission.
- *
- * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
- * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
- * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
- * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
- * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
- * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
- * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
- * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
- * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
- * POSSIBILITY OF SUCH DAMAGE.
- */
-
-/dts-v1/;
-/ {
-	dpl-version = <10>;
-
-	containers {
-		dprc@1 {
-			compatible = "fsl,dprc";
-			parent = "none";
-			options = "DPRC_CFG_OPT_SPAWN_ALLOWED", "DPRC_CFG_OPT_ALLOC_ALLOWED", "DPRC_CFG_OPT_OBJ_CREATE_ALLOWED", "DPRC_CFG_OPT_TOPOLOGY_CHANGES_ALLOWED", "DPRC_CFG_OPT_IRQ_CFG_ALLOWED";
-
-			objects {
-				obj_set@dpmac {
-					type = "dpmac";
-					ids = <0xc 0xd 0xe 0x10 0x11 0x12>;
-				};
-
-				obj_set@dpmcp {
-					type = "dpmcp";
-					ids = <0x1 0x2 0x3 0x4 0x5 0x6 0x7 0x8 0x9 0xa 0xb 0xc 0xd 0xe 0xf 0x10 0x11 0x12 0x13 0x14 0x15 0x16 0x17 0x18 0x19 0x1a 0x1b 0x1c 0x1d 0x1e 0x1f 0x20 0x21 0x22 0x23>;
-				};
-			};
-		};
-	};
-
-	objects {
-		/* Serdes 2 */
-		dpmac@12 {
-		};
-		dpmac@13 {
-		};
-		dpmac@14 {
-		};
-		dpmac@16 {
-		};
-		dpmac@17 {
-		};
-		dpmac@18 {
-		};
-
-		dpmcp@1 {
-		};
-		dpmcp@2 {
-		};
-		dpmcp@3 {
-		};
-		dpmcp@4 {
-		};
-		dpmcp@5 {
-		};
-		dpmcp@6 {
-		};
-		dpmcp@7 {
-		};
-		dpmcp@8 {
-		};
-		dpmcp@9 {
-		};
-		dpmcp@10 {
-		};
-		dpmcp@11 {
-		};
-		dpmcp@12 {
-		};
-		dpmcp@13 {
-		};
-		dpmcp@14 {
-		};
-		dpmcp@15 {
-		};
-		dpmcp@16 {
-		};
-		dpmcp@17 {
-		};
-		dpmcp@18 {
-		};
-		dpmcp@19 {
-		};
-		dpmcp@20 {
-		};
-		dpmcp@21 {
-		};
-		dpmcp@22 {
-		};
-		dpmcp@23 {
-		};
-		dpmcp@24 {
-		};
-		dpmcp@25 {
-		};
-		dpmcp@26 {
-		};
-		dpmcp@27 {
-		};
-		dpmcp@28 {
-		};
-		dpmcp@29 {
-		};
-		dpmcp@30 {
-		};
-		dpmcp@31 {
-		};
-		dpmcp@32 {
-		};
-		dpmcp@33 {
-		};
-		dpmcp@34 {
-		};
-		dpmcp@35 {
-		};
-	};
-
-	connections {
-	};
-};
diff --git a/config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_7-dpc.dts b/config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_7-dpc.dts
deleted file mode 100644
index b199f64..0000000
--- a/config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_7-dpc.dts
+++ /dev/null
@@ -1,102 +0,0 @@
-/*
- * Copyright 2020 NXP
- * Copyright 2022 Josua Mayer <josua@solid-run.com>
- *
- * Redistribution and use in source and binary forms, with or without
- * modification, are permitted provided that the following conditions are met:
- *   * Redistributions of source code must retain the above copyright
- *     notice, this list of conditions and the following disclaimer.
- *   * Redistributions in binary form must reproduce the above copyright
- *     notice, this list of conditions and the following disclaimer in the
- *     documentation and/or other materials provided with the distribution.
- *   * Neither the name of the above-listed copyright holders nor the
- *     names of any contributors may be used to endorse or promote products
- *     derived from this software without specific prior written permission.
- *
- * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
- * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
- * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
- * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
- * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
- * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
- * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
- * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
- * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
- * POSSIBILITY OF SUCH DAMAGE.
- */
-
-/dts-v1/;
-
-/ {
-	resources {
-		icid_pools {
-			icid_pool@1 {
-				num = <0x64>;
-				base_icid = <0x0>;
-			};
-		};
-	};
-
-	mc_general {
-		log {
-			mode = "LOG_MODE_ON";
-			level = "LOG_LEVEL_WARNING";
-		};
-
-		console {
-			mode = "CONSOLE_MODE_OFF";
-			uart_id = <0x4>;
-			level = "LOG_LEVEL_WARNING";
-		};
-	};
-
-	controllers {
-		qbman {
-			/* Transform this number of 8-WQ channels into four times
-			 * as many 2-WQ channels. This allows the creation of a
-			 * larger number of DPCONs.
-			 */
-			wq_ch_conversion = <64>;
-		};
-	};
-
-	board_info {
-		recycle_ports {
-			recycle@1 {
-				max_rate = "1G";
-			};
-
-			recycle@2 {
-				max_rate = "1G";
-			};
-		};
-
-		ports {
-			/* Serdes 2 */
-			mac@12 {
-				link_type = "MAC_LINK_TYPE_PHY";
-			};
-
-			mac@13 {
-				link_type = "MAC_LINK_TYPE_PHY";
-			};
-
-			mac@14 {
-				link_type = "MAC_LINK_TYPE_PHY";
-			};
-
-			mac@16 {
-				link_type = "MAC_LINK_TYPE_PHY";
-			};
-
-			mac@17 {
-				link_type = "MAC_LINK_TYPE_PHY";
-			};
-
-			mac@18 {
-				link_type = "MAC_LINK_TYPE_PHY";
-			};
-		};
-	};
-};
diff --git a/config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_7-dpl.dts b/config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_7-dpl.dts
deleted file mode 100644
index 766b04d..0000000
--- a/config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_7-dpl.dts
+++ /dev/null
@@ -1,142 +0,0 @@
-/*
- * Copyright 2020 NXP
- * Copyright 2022 Josua Mayer <josua@solid-run.com>
- *
- * Redistribution and use in source and binary forms, with or without
- * modification, are permitted provided that the following conditions are met:
- *   * Redistributions of source code must retain the above copyright
- *     notice, this list of conditions and the following disclaimer.
- *   * Redistributions in binary form must reproduce the above copyright
- *     notice, this list of conditions and the following disclaimer in the
- *     documentation and/or other materials provided with the distribution.
- *   * Neither the name of the above-listed copyright holders nor the
- *     names of any contributors may be used to endorse or promote products
- *     derived from this software without specific prior written permission.
- *
- * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
- * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
- * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
- * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
- * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
- * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
- * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
- * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
- * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
- * POSSIBILITY OF SUCH DAMAGE.
- */
-
-/dts-v1/;
-/ {
-	dpl-version = <10>;
-
-	containers {
-		dprc@1 {
-			compatible = "fsl,dprc";
-			parent = "none";
-			options = "DPRC_CFG_OPT_SPAWN_ALLOWED", "DPRC_CFG_OPT_ALLOC_ALLOWED", "DPRC_CFG_OPT_OBJ_CREATE_ALLOWED", "DPRC_CFG_OPT_TOPOLOGY_CHANGES_ALLOWED", "DPRC_CFG_OPT_IRQ_CFG_ALLOWED";
-
-			objects {
-				obj_set@dpmac {
-					type = "dpmac";
-					ids = <0xc 0xd 0xe 0x10 0x11 0x12>;
-				};
-
-				obj_set@dpmcp {
-					type = "dpmcp";
-					ids = <0x1 0x2 0x3 0x4 0x5 0x6 0x7 0x8 0x9 0xa 0xb 0xc 0xd 0xe 0xf 0x10 0x11 0x12 0x13 0x14 0x15 0x16 0x17 0x18 0x19 0x1a 0x1b 0x1c 0x1d 0x1e 0x1f 0x20 0x21 0x22 0x23>;
-				};
-			};
-		};
-	};
-
-	objects {
-		/* Serdes 2 */
-		dpmac@12 {
-		};
-		dpmac@13 {
-		};
-		dpmac@14 {
-		};
-		dpmac@16 {
-		};
-		dpmac@17 {
-		};
-		dpmac@18 {
-		};
-
-		dpmcp@1 {
-		};
-		dpmcp@2 {
-		};
-		dpmcp@3 {
-		};
-		dpmcp@4 {
-		};
-		dpmcp@5 {
-		};
-		dpmcp@6 {
-		};
-		dpmcp@7 {
-		};
-		dpmcp@8 {
-		};
-		dpmcp@9 {
-		};
-		dpmcp@10 {
-		};
-		dpmcp@11 {
-		};
-		dpmcp@12 {
-		};
-		dpmcp@13 {
-		};
-		dpmcp@14 {
-		};
-		dpmcp@15 {
-		};
-		dpmcp@16 {
-		};
-		dpmcp@17 {
-		};
-		dpmcp@18 {
-		};
-		dpmcp@19 {
-		};
-		dpmcp@20 {
-		};
-		dpmcp@21 {
-		};
-		dpmcp@22 {
-		};
-		dpmcp@23 {
-		};
-		dpmcp@24 {
-		};
-		dpmcp@25 {
-		};
-		dpmcp@26 {
-		};
-		dpmcp@27 {
-		};
-		dpmcp@28 {
-		};
-		dpmcp@29 {
-		};
-		dpmcp@30 {
-		};
-		dpmcp@31 {
-		};
-		dpmcp@32 {
-		};
-		dpmcp@33 {
-		};
-		dpmcp@34 {
-		};
-		dpmcp@35 {
-		};
-	};
-
-	connections {
-	};
-};
diff --git a/config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_0-dpc.dts b/config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_0-dpc.dts
deleted file mode 100644
index db6f440..0000000
--- a/config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_0-dpc.dts
+++ /dev/null
@@ -1,94 +0,0 @@
-/*
- * Copyright 2020 NXP
- * Copyright 2022 Josua Mayer <josua@solid-run.com>
- *
- * Redistribution and use in source and binary forms, with or without
- * modification, are permitted provided that the following conditions are met:
- *   * Redistributions of source code must retain the above copyright
- *     notice, this list of conditions and the following disclaimer.
- *   * Redistributions in binary form must reproduce the above copyright
- *     notice, this list of conditions and the following disclaimer in the
- *     documentation and/or other materials provided with the distribution.
- *   * Neither the name of the above-listed copyright holders nor the
- *     names of any contributors may be used to endorse or promote products
- *     derived from this software without specific prior written permission.
- *
- * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
- * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
- * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
- * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
- * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
- * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
- * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
- * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
- * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
- * POSSIBILITY OF SUCH DAMAGE.
- */
-
-/dts-v1/;
-
-/ {
-	resources {
-		icid_pools {
-			icid_pool@1 {
-				num = <0x64>;
-				base_icid = <0x0>;
-			};
-		};
-	};
-
-	mc_general {
-		log {
-			mode = "LOG_MODE_ON";
-			level = "LOG_LEVEL_WARNING";
-		};
-
-		console {
-			mode = "CONSOLE_MODE_OFF";
-			uart_id = <0x4>;
-			level = "LOG_LEVEL_WARNING";
-		};
-	};
-
-	controllers {
-		qbman {
-			/* Transform this number of 8-WQ channels into four times
-			 * as many 2-WQ channels. This allows the creation of a
-			 * larger number of DPCONs.
-			 */
-			wq_ch_conversion = <64>;
-		};
-	};
-
-	board_info {
-		recycle_ports {
-			recycle@1 {
-				max_rate = "1G";
-			};
-
-			recycle@2 {
-				max_rate = "1G";
-			};
-		};
-
-		ports {
-			/* Serdes 1 */
-			mac@3 {
-				link_type = "MAC_LINK_TYPE_PHY";
-			};
-
-			mac@4 {
-				link_type = "MAC_LINK_TYPE_PHY";
-			};
-
-			mac@5 {
-				link_type = "MAC_LINK_TYPE_PHY";
-			};
-
-			mac@6 {
-				link_type = "MAC_LINK_TYPE_PHY";
-			};
-		};
-	};
-};
diff --git a/config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_0-dpl.dts b/config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_0-dpl.dts
deleted file mode 100644
index b76d2c1..0000000
--- a/config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_0-dpl.dts
+++ /dev/null
@@ -1,138 +0,0 @@
-/*
- * Copyright 2020 NXP
- * Copyright 2022 Josua Mayer <josua@solid-run.com>
- *
- * Redistribution and use in source and binary forms, with or without
- * modification, are permitted provided that the following conditions are met:
- *   * Redistributions of source code must retain the above copyright
- *     notice, this list of conditions and the following disclaimer.
- *   * Redistributions in binary form must reproduce the above copyright
- *     notice, this list of conditions and the following disclaimer in the
- *     documentation and/or other materials provided with the distribution.
- *   * Neither the name of the above-listed copyright holders nor the
- *     names of any contributors may be used to endorse or promote products
- *     derived from this software without specific prior written permission.
- *
- * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
- * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
- * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
- * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
- * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
- * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
- * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
- * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
- * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
- * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
- * POSSIBILITY OF SUCH DAMAGE.
- */
-
-/dts-v1/;
-/ {
-	dpl-version = <10>;
-
-	containers {
-		dprc@1 {
-			compatible = "fsl,dprc";
-			parent = "none";
-			options = "DPRC_CFG_OPT_SPAWN_ALLOWED", "DPRC_CFG_OPT_ALLOC_ALLOWED", "DPRC_CFG_OPT_OBJ_CREATE_ALLOWED", "DPRC_CFG_OPT_TOPOLOGY_CHANGES_ALLOWED", "DPRC_CFG_OPT_IRQ_CFG_ALLOWED";
-
-			objects {
-				obj_set@dpmac {
-					type = "dpmac";
-					ids = <0x3 0x4 0x5 0x6>;
-				};
-
-				obj_set@dpmcp {
-					type = "dpmcp";
-					ids = <0x1 0x2 0x3 0x4 0x5 0x6 0x7 0x8 0x9 0xa 0xb 0xc 0xd 0xe 0xf 0x10 0x11 0x12 0x13 0x14 0x15 0x16 0x17 0x18 0x19 0x1a 0x1b 0x1c 0x1d 0x1e 0x1f 0x20 0x21 0x22 0x23>;
-				};
-			};
-		};
-	};
-
-	objects {
-		/* Serdes 1 */
-		dpmac@3 {
-		};
-		dpmac@4 {
-		};
-		dpmac@5 {
-		};
-		dpmac@6 {
-		};
-
-		dpmcp@1 {
-		};
-		dpmcp@2 {
-		};
-		dpmcp@3 {
-		};
-		dpmcp@4 {
-		};
-		dpmcp@5 {
-		};
-		dpmcp@6 {
-		};
-		dpmcp@7 {
-		};
-		dpmcp@8 {
-		};
-		dpmcp@9 {
-		};
-		dpmcp@10 {
-		};
-		dpmcp@11 {
-		};
-		dpmcp@12 {
-		};
-		dpmcp@13 {
-		};
-		dpmcp@14 {
-		};
-		dpmcp@15 {
-		};
-		dpmcp@16 {
-		};
-		dpmcp@17 {
-		};
-		dpmcp@18 {
-		};
-		dpmcp@19 {
-		};
-		dpmcp@20 {
-		};
-		dpmcp@21 {
-		};
-		dpmcp@22 {
-		};
-		dpmcp@23 {
-		};
-		dpmcp@24 {
-		};
-		dpmcp@25 {
-		};
-		dpmcp@26 {
-		};
-		dpmcp@27 {
-		};
-		dpmcp@28 {
-		};
-		dpmcp@29 {
-		};
-		dpmcp@30 {
-		};
-		dpmcp@31 {
-		};
-		dpmcp@32 {
-		};
-		dpmcp@33 {
-		};
-		dpmcp@34 {
-		};
-		dpmcp@35 {
-		};
-	};
-
-	connections {
-	};
-};
diff --git a/config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_9-dpc.dts b/config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_7-dpc.dts
similarity index 94%
rename from config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_9-dpc.dts
rename to config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_7-dpc.dts
index c780130..0f0ec4b 100644
--- a/config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_9-dpc.dts
+++ b/config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_7-dpc.dts
@@ -73,24 +73,33 @@
 		};
 
 		ports {
-			/* Serdes 2 */
-			mac@11 {
+			/* Serdes 1 */
+			mac@3 {
 				link_type = "MAC_LINK_TYPE_PHY";
 			};
 
-			mac@12 {
+			mac@4 {
 				link_type = "MAC_LINK_TYPE_PHY";
 			};
 
-			mac@13 {
+			mac@5 {
 				link_type = "MAC_LINK_TYPE_PHY";
 			};
 
-			mac@14 {
+			mac@6 {
+				link_type = "MAC_LINK_TYPE_PHY";
+			};
+
+			/* Serdes 2 */
+			mac@12 {
 				link_type = "MAC_LINK_TYPE_PHY";
 			};
 
-			mac@15 {
+			mac@13 {
+				link_type = "MAC_LINK_TYPE_PHY";
+			};
+
+			mac@14 {
 				link_type = "MAC_LINK_TYPE_PHY";
 			};
 
diff --git a/config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_9-dpl.dts b/config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_7-dpl.dts
similarity index 64%
rename from config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_9-dpl.dts
rename to config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_7-dpl.dts
index 066238a..47a123a 100644
--- a/config/lx2160a/LX2162-USOM/clearfog-s1_0-s2_9-dpl.dts
+++ b/config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_7-dpl.dts
@@ -31,6 +31,7 @@
 	dpl-version = <10>;
 
 	containers {
+
 		dprc@1 {
 			compatible = "fsl,dprc";
 			parent = "none";
@@ -39,105 +40,200 @@
 			objects {
 				obj_set@dpmac {
 					type = "dpmac";
-					ids = <0xb 0xc 0xd 0xe 0xf 0x10 0x11 0x12>;
+					ids = <3 4 5 6 12 13 14 16 17 18 >;
 				};
 
 				obj_set@dpmcp {
 					type = "dpmcp";
-					ids = <0x1 0x2 0x3 0x4 0x5 0x6 0x7 0x8 0x9 0xa 0xb 0xc 0xd 0xe 0xf 0x10 0x11 0x12 0x13 0x14 0x15 0x16 0x17 0x18 0x19 0x1a 0x1b 0x1c 0x1d 0x1e 0x1f 0x20 0x21 0x22 0x23>;
+					ids = <1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 >;
 				};
 			};
 		};
 	};
 
 	objects {
-		/* Serdes 2 */
-		dpmac@11 {
+
+		/* Serdes 1 */
+		dpmac@3 {
+			compatible = "fsl,dpmac";
+		};
+
+		dpmac@4 {
+			compatible = "fsl,dpmac";
+		};
+
+		dpmac@5 {
+			compatible = "fsl,dpmac";
 		};
+
+		dpmac@6 {
+			compatible = "fsl,dpmac";
+		};
+
+		/* Serdes 2 */
+
 		dpmac@12 {
+			compatible = "fsl,dpmac";
 		};
+
 		dpmac@13 {
+			compatible = "fsl,dpmac";
 		};
+
 		dpmac@14 {
+			compatible = "fsl,dpmac";
 		};
-		dpmac@15 {
-		};
+
 		dpmac@16 {
+			compatible = "fsl,dpmac";
 		};
+
 		dpmac@17 {
+			compatible = "fsl,dpmac";
 		};
+
 		dpmac@18 {
+			compatible = "fsl,dpmac";
 		};
 
 		dpmcp@1 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@2 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@3 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@4 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@5 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@6 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@7 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@8 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@9 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@10 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@11 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@12 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@13 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@14 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@15 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@16 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@17 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@18 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@19 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@20 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@21 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@22 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@23 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@24 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@25 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@26 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@27 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@28 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@29 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@30 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@31 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@32 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@33 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@34 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@35 {
+			compatible = "fsl,dpmcp";
 		};
 	};
 
diff --git a/config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_9-dpl.dts b/config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_9-dpl.dts
index efe1c60..cf8feb3 100644
--- a/config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_9-dpl.dts
+++ b/config/lx2160a/LX2162-USOM/clearfog-s1_3-s2_9-dpl.dts
@@ -31,6 +31,7 @@
 	dpl-version = <10>;
 
 	containers {
+
 		dprc@1 {
 			compatible = "fsl,dprc";
 			parent = "none";
@@ -39,116 +40,207 @@
 			objects {
 				obj_set@dpmac {
 					type = "dpmac";
-					ids = <0x3 0x4 0x5 0x6 0xb 0xc 0xd 0xe 0xf 0x10 0x11 0x12>;
+					ids = <3 4 5 6 11 12 13 14 15 16 17 18 >;
 				};
 
-				/* 1x per dpni + 1x per dpmac */
 				obj_set@dpmcp {
 					type = "dpmcp";
-					ids = <0x1 0x2 0x3 0x4 0x5 0x6 0x7 0x8 0x9 0xa 0xb 0xc 0xd 0xe 0xf 0x10 0x11 0x12 0x13 0x14 0x15 0x16 0x17 0x18 0x19 0x1a 0x1b 0x1c 0x1d 0x1e 0x1f 0x20 0x21 0x22 0x23>;
+					ids = <1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 >;
 				};
 			};
 		};
 	};
 
 	objects {
+
 		/* Serdes 1 */
 		dpmac@3 {
+			compatible = "fsl,dpmac";
 		};
+
 		dpmac@4 {
+			compatible = "fsl,dpmac";
 		};
+
 		dpmac@5 {
+			compatible = "fsl,dpmac";
 		};
+
 		dpmac@6 {
+			compatible = "fsl,dpmac";
 		};
 
 		/* Serdes 2 */
 		dpmac@11 {
+			compatible = "fsl,dpmac";
 		};
+
 		dpmac@12 {
+			compatible = "fsl,dpmac";
 		};
+
 		dpmac@13 {
+			compatible = "fsl,dpmac";
 		};
+
 		dpmac@14 {
+			compatible = "fsl,dpmac";
 		};
+
 		dpmac@15 {
+			compatible = "fsl,dpmac";
 		};
+
 		dpmac@16 {
+			compatible = "fsl,dpmac";
 		};
+
 		dpmac@17 {
+			compatible = "fsl,dpmac";
 		};
+
 		dpmac@18 {
+			compatible = "fsl,dpmac";
 		};
 
 		dpmcp@1 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@2 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@3 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@4 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@5 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@6 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@7 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@8 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@9 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@10 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@11 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@12 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@13 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@14 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@15 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@16 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@17 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@18 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@19 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@20 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@21 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@22 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@23 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@24 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@25 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@26 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@27 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@28 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@29 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@30 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@31 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@32 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@33 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@34 {
+			compatible = "fsl,dpmcp";
 		};
+
 		dpmcp@35 {
+			compatible = "fsl,dpmcp";
 		};
 	};
 
-- 
2.35.3

